MekHQ/src/mekhq/campaign/autoresolve/acar/report/EndPhaseReporter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.autoresolve.acar.report;
+
+import megamek.common.Entity;
+import megamek.common.IEntityRemovalConditions;
+import megamek.common.IGame;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class EndPhaseReporter {
+
+    private final Consumer<PublicReportEntry> reportConsumer;
+    private static final Map<Integer, Integer> unitDestroyedMessageMap = Map.of(
+        IEntityRemovalConditions.REMOVE_DEVASTATED, 3337,
```

Comment:
Using integers like this is really confusing, at least looking at it as a reviewer. If these integers parse to a specific meaning, would there be value in converting them into enums, instead?

Reply:
Those are messageIDs for the gamelogger. It is just a reference map saying "this number refers to that other number" so it knows which report to load in the acs-report-message-LOCALE.properties, this was the best I could do since I rely on the IEntityRemovalConditions due to the DamageApplier class.